src/tracking-buffer/writable-tracking-buffer.js

```diff
@@ -52,14 +53,13 @@ module.exports = class WritableTrackingBuffer {
   }
 
   newBuffer(size) {
-    size || (size = this.initialSize);
     const buffer = this.buffer.slice(0, this.position);
     if (this.compositeBuffer) {
       this.compositeBuffer = Buffer.concat([this.compositeBuffer, buffer]);
     } else {
       this.compositeBuffer = buffer;
     }
-    this.buffer = new Buffer(size);
+    this.buffer = (size === 0) ? ZERO_LENGTH_BUFFER : new Buffer(size);
```

Comment:
Yes, ZERO_LENGGTH_BUFFER is common for all instances. But it's a zero length buffer and therefore immutable.